nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/test/java/org/apache/nifi/processors/elasticsearch/PutElasticsearchRecordTest.java

```diff
@@ -0,0 +1,850 @@
+        runner.setProperty(PutElasticsearchRecord.NOT_FOUND_IS_SUCCESSFUL, "false");
+        clientService.setResponse(IndexOperationResponse.fromJsonResponse(Files.readString(Paths.get(TEST_COMMON_DIR, "sampleErrorResponse.json"))));
+        registry.addSchema(ERROR_TEST_SCHEMA, errorTestSchema);
+        runner.enqueue(VALUES, Collections.singletonMap("schema.name", ERROR_TEST_SCHEMA));
+        runner.assertValid();
+        runner.run();
+
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESS, 1);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_RETRY, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILED_RECORDS, 1);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESSFUL_RECORDS, 1);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_ERROR_RESPONSES, 0);
+
+        runner.getFlowFilesForRelationship(PutElasticsearchRecord.REL_FAILED_RECORDS).get(0).assertAttributeEquals(PutElasticsearchRecord.ATTR_RECORD_COUNT, "2");
+        runner.getFlowFilesForRelationship(PutElasticsearchRecord.REL_SUCCESSFUL_RECORDS).get(0).assertAttributeEquals(PutElasticsearchRecord.ATTR_RECORD_COUNT, "3");
+
+        assertEquals(1,
+                runner.getProvenanceEvents().stream().filter(
+                        e -> ProvenanceEventType.SEND.equals(e.getEventType()) && "1 Elasticsearch _bulk operation batch(es) [2 error(s), 3 success(es)]".equals(e.getDetails())).count());
+    }
+
+    @Test
+    public void testDocumentNotFoundErrorHandling() throws Exception {
+        runner.setProperty(PutElasticsearchRecord.NOT_FOUND_IS_SUCCESSFUL, "false");
+        runner.setProperty(PutElasticsearchRecord.OUTPUT_ERROR_RESPONSES, "true");
+        clientService.setResponse(IndexOperationResponse.fromJsonResponse(Files.readString(Paths.get(TEST_COMMON_DIR, "sampleErrorResponse.json"))));
+        registry.addSchema(ERROR_TEST_SCHEMA, errorTestSchema);
+        runner.enqueue(VALUES, Collections.singletonMap("schema.name", ERROR_TEST_SCHEMA));
+        runner.assertValid();
+        runner.run();
+
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESS, 1);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILURE, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_RETRY, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_FAILED_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_SUCCESSFUL_RECORDS, 0);
+        runner.assertTransferCount(PutElasticsearchRecord.REL_ERROR_RESPONSES, 1);
+
+        final String errorResponses = runner.getFlowFilesForRelationship(PutElasticsearchJson.REL_ERROR_RESPONSES).get(0).getContent();
+        //assertThat(errorResponses, containsString("not_found"))
+        assertTrue(errorResponses.contains("not_found"));
+        //assertThat(errorResponses, containsString("For input string: 20abc"))
```

Comment:
```suggestion
        assertTrue(errorResponses.contains("not_found"));
```

Remove unnecessary commented out code